### DISCUSSION TITLE
Same image has multiple rows in metadata but only single is displayed in viewer

### DISCUSSION THREAD

**Discussion by SushantGautam** [huggingface/datasets#7131] [Ideas]
Need feature to show multiple rows referring to the same images in dataset viewer.

For eg in VQA dataset.
I have 10K images and say 100K Q-A annotations.
But only 10K data rows are shown in the viewer. .  seems like it selected one row per unique filename.
any way to address this?

I tried with dataset loader script, but that was not so efficient. 
Due to multiple images the size got huge.
Need help!

Metadata sample:
file_name,question,answer
a.jpg,what is this?Animal
a.jpg,what is this?Vertebrate
a.jpg,what is this?Cow
b.jpg,what is tasty?Grass

Point->> This is perceived as two rows by the viewer.

**Comment by dexhunter** [comment]
Current `imagefolder` supports multiple metadata rows referring to the same image. I checked this with `datasets==5.0.1`: two image files and your four question/answer records produce four dataset rows.

Use this layout:

```text
data/
├── a.jpg
├── b.jpg
└── metadata.jsonl
```

```jsonl
{"file_name":"a.jpg","question":"what is this?","answer":"Animal"}
{"file_name":"a.jpg","question":"what is this?","answer":"Vertebrate"}
{"file_name":"a.jpg","question":"what is this?","answer":"Cow"}
{"file_name":"b.jpg","question":"what is tasty?","answer":"Grass"}
```

```python
from datasets import load_dataset

ds = load_dataset("imagefolder", data_dir="data", split="train")
print(len(ds))  # 4
```

Upload the image files and `metadata.jsonl` to the dataset repository, preserving those relative paths. This uses the [supported ImageFolder metadata format](https://huggingface.co/docs/datasets/image_dataset#imagefolder), so no custom loading script is needed. Your CSV sample also appears to be missing the comma between each question and answer; JSONL avoids that ambiguity.

This keeps only one source copy of each image in the repository. It does not guarantee deduplication in the viewer's generated Parquet files. If the hosted viewer still shows two rows with this layout, the repository URL would be needed to investigate its conversion separately from the local loader.
